AutoNumber/BlobOptimisticDataStore.cs

```diff
@@ -78,43 +116,64 @@ public async Task<bool> TryOptimisticWriteAsync(string blockName, string data)
             return true;
         }
 
-        private ICloudBlob GetBlobReference(string blockName)
+        private BlockBlobClient GetBlobReference(string blockName)
         {
             return blobReferences.GetValue(
                 blockName,
                 blobReferencesLock,
-                () => InitializeBlobReferenceAsync(blockName).GetAwaiter().GetResult());
+                () => InitializeBlobReference(blockName));
         }
 
-        private async Task<ICloudBlob> InitializeBlobReferenceAsync(string blockName)
+        private BlockBlobClient InitializeBlobReference(string blockName)
         {
-            var blobReference = blobContainer.GetBlockBlobReference(blockName);
+            var blobReference = blobContainer.GetBlockBlobClient(blockName);
 
-            if (await blobReference.ExistsAsync().ConfigureAwait(false))
+            if (blobReference.Exists())
                 return blobReference;
 
             try
             {
-                await UploadTextAsync(blobReference, SeedValue, AccessCondition.GenerateIfNoneMatchCondition("*"))
-                    .ConfigureAwait(false);
+                var blobRequestCondition = new BlobRequestConditions
+                {
+                    IfNoneMatch = ETag.All
+                };
+                UploadText(blobReference, SeedValue, blobRequestCondition);
             }
-            catch (StorageException uploadException)
+            catch (RequestFailedException uploadException)
             {
-                if (uploadException.RequestInformation.HttpStatusCode != (int) HttpStatusCode.Conflict)
+                if (uploadException.Status != (int)HttpStatusCode.Conflict)
                     throw;
             }
 
             return blobReference;
         }
 
-        private async Task UploadTextAsync(ICloudBlob blob, string text, AccessCondition accessCondition)
+        private async Task UploadTextAsync(BlockBlobClient blob, string text, BlobRequestConditions accessCondition)
+        {
+            //blob.Properties.ContentType = "utf-8";
+            //blob.Properties.ContentType = "text/plain";
+
+            var header = new BlobHttpHeaders
+            {
+                ContentType = "text/plain"
+            };
+
+            using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(text)))
+            {
+                await blob.UploadAsync(stream, header, null, accessCondition, null, null).ConfigureAwait(false);
+            }
+        }
+
+        private void UploadText(BlockBlobClient blob, string text, BlobRequestConditions accessCondition)
         {
-            blob.Properties.ContentType = "utf-8";
-            blob.Properties.ContentType = "text/plain";

```

Comment:
I'm not sure what this was doing since it appears to me that the second one "text/plain" is overriding the first one, what was the purpose of this?

Reply:
I don't idea behind this, I ported this code from original source.

Reply:
Kept just text/plain